TKeazirian.HTTPServer/Server/Router.cs

```diff
@@ -56,4 +64,31 @@ private Response HeadRequest(Handler handler, Request request)
         );
         return headResponse;
     }
+
+    private Response OptionsRequest(Request request)
+    {
+        var optionsResponse = new ResponseBuilder()
+            .SetStatusCode(HttpStatusCode.Ok)
+            .SetHeaders("Allow", AddToAllowedMethodsForOptions(request))
+            .Build();
+
+        return optionsResponse;
+    }
+
+    public string GetAllowedMethodsFromHandler(Handler handler)
+    {
+        List<string> allowedMethods = handler.AllowedHttpMethods();
+
+        string allowedMethodsString = string.Join(", ", allowedMethods);
+
+        return allowedMethodsString;
+    }
+
+    public string AddToAllowedMethodsForOptions(Request request)
+    {
+        string allowedMethods = GetAllowedMethodsFromHandler(_routesConfig.Routes[request.GetRequestPath()]);
+
+        string additionalAllowedMethods = "HEAD, OPTIONS";
+        return $"{allowedMethods}, {additionalAllowedMethods}";
+    }
```

Comment:
what happens if GET is not implemented on an endpoint?  should you still "get" HEAD & OPTIONS?  If not, should there be something enforcing that at a minimum all endpoints have a GET?

Reply:
... even if that enforcing is just an update to the README?

Reply:
Well, if GET is not implemented, can you have HEAD or OPTIONS?  Since you get them "for free" with GET, if they don't have GET, then do you still get them?

Reply:
I think you have your answer.

Reply:
Picking back up from my other comments below. What would be the purpose of adding GET to all endpoints? In theory, should you be able to have an endpoint that doesn't serve GET requests? What I've been trying to get at is that I would expect an OPTIONS request to return ALL of the implemented HTTP methods for ANY given endpoint in the server. So if an endpoint only has POST and PUT configured what would you expect the OPTIONS response to that endpoint to be? Should that be in scope for this ticket?

Reply:
So, what I'm asking is: why is GET privileged here? HEAD and OPTIONS are two different types of requests, so I think you need to answer two questions:

Does an endpoint need to serve GET requests to respond to a HEAD request?

Does an endpoint need to serve GET requests to respond to an OPTIONS request?

My understanding is that when you have a GET you get HEAD "for free," but you get OPTIONS "for free" for any endpoint that is configured with any methods, whether or not those methods include GET.

Reply:
https://stackoverflow.com/questions/60232424/using-http-head-method-when-the-underlying-ressource-does-not-allow-get

Reply:
I see!  Okay that definitely makes sense.  Let me take a look at this from that perspective!